src/api/pricer/CoinGecko.ts

```diff
@@ -0,0 +1,65 @@
+import type { ResponseError } from 'superagent';
+import { BigNumber as BigNumberJs } from 'bignumber.js';
+import BaseExchangeApi, { CurrencyMapping } from './ExchangeApi';
+import HttpWrapper from '../common/HttpWrapper';
+
+const BASE_URL = 'https://api.coingecko.com/';
+const PRICE_API_PATH = '/api/v3/simple/price';
+
+export type CoinGeckoResponse = Record<string, Record<string, string>>;
+
+export const COINGECKO_RBTC_ID = 'rootstock';
+export const COINGECKO_RIF_TOKEN_ID = 'rif-token';
+
+const CURRENCY_MAPPING: CurrencyMapping = {
+  RIF: COINGECKO_RIF_TOKEN_ID,
+  RBTC: COINGECKO_RBTC_ID,
+  TRIF: COINGECKO_RIF_TOKEN_ID,
+};
+
+export default class CoinGecko extends BaseExchangeApi {
+  constructor(
+    private readonly _httpWrapper: HttpWrapper = new HttpWrapper(),
+    private readonly _baseUrl = BASE_URL
+  ) {
+    super('CoinGecko', CURRENCY_MAPPING, ['RIF', 'RBTC', 'tRif']);
+  }
+
+  async queryExchangeRate(
+    sourceCurrency: string,
+    targetCurrency: string
+  ): Promise<BigNumberJs> {
+    let response: CoinGeckoResponse;
+    // TODO: We could check if the target currency is supported by calling the /simple/supported_vs_currencies API
```

Comment:
I didn't implement it for 2 main reasons:
- right now we use it with `usd` only
- I didn't want to make an additional call if not strictly necessary, since there is a rate limit for the free plan